co-authors-plus.php

```diff
@@ -1052,22 +1052,61 @@ function filter_count_user_posts( $count, $user_id ) {
 
 		if ( is_object( $user ) ) {
 			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
-
 			$term = $this->get_author_term( $user );
 
-			if ( $term && ! is_wp_error( $term ) ) {
-				if ( 'guest-author' === $user->type ) {
-					// If using guest author term count, add on linked user count.
-					$count = (int) $count + $term->count;
-				} else {
-					$count = $term->count;
-				}
+			if ( strlen( $user->linked_account ) > 1 ) {
```

Comment:
What if `$user` does not return an object from line 1054?

Reply:
> What if $user does not return an object from line 1054?
In this case, my fix would probably trigger an error. That said, I have rewritten the that function from scratch and added inline comments to it, so that others understand how the posts are counted.